### PR TITLE
[Sofa] Compilation without Compatibility mode

### PR DESCRIPTION
**Pull request by fredroy**
Based on 
 - #2958 

Contains:
- #2960 

Allow compilation of the Sofa codebase without the cmake option SOFA_ENABLE_LEGACY_HEADERS, i.e the compatibility mode.

(Sofa codebase = default cmake options, i.e framework, Component, tests and the few plugins ON by default)

[NOTci-depends-on https://github.com/sofa-framework/Registration/pull/2]
[NOTci-depends-on https://github.com/sofa-framework/SofaPython3/pull/268]
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
@fredroy you have an error on Windows

File: SofaKernel/modules/Sofa.GL/Sofa.GLConfig.cmake.in [hunk 1]
```diff
@@ -5,9 +5,10 @@
 
 set(SOFA_GL_HAVE_GLEW @SOFA_GL_HAVE_GLEW@)
 
-find_package(SofaFramework QUIET REQUIRED) # Sofa.Helper Sofa.DefaultType
 find_package(OpenGL QUIET REQUIRED)
 find_package(GLEW QUIET REQUIRED)
+find_package(Sofa.Helper REQUIRED)
+find_package(Sofa.DefaultType REQUIRED)
```
Comment (guparan) [inline SofaKernel/modules/Sofa.GL/Sofa.GLConfig.cmake.in]:
```suggestion
find_package(Sofa.Helper QUIET REQUIRED)
find_package(Sofa.DefaultType QUIET REQUIRED)
```

File: applications/plugins/ArticulatedSystemPlugin/ArticulatedSystemPluginConfig.cmake.in [hunk 2]
```diff
@@ -3,17 +3,7 @@
...
-
-find_package(SofaBase QUIET REQUIRED)
-
-if(PLUGINEXAMPLE_HAVE_SOFAGUI)
-    find_package(SofaGui QUIET REQUIRED)
-endif()
-if(PLUGINEXAMPLE_HAVE_QT5CORE)
-    find_package(Qt5 COMPONENTS Core QUIET REQUIRED)
-endif()
+find_package(Sofa.Simulation.Core QUIET REQUIRED)
```
Comment (guparan) [inline applications/plugins/ArticulatedSystemPlugin/ArticulatedSystemPluginConfig.cmake.in:6]:
```suggestion
find_package(Sofa.Simulation.Core QUIET REQUIRED)
find_package(Sofa.Component.Controller QUIET REQUIRED)
```

File: applications/plugins/SofaTest/SofaTestConfig.cmake.in [hunk 3]
```diff
@@ -3,14 +3,10 @@
...
-find_package(SofaMiscMapping REQUIRED)
-find_package(SofaMiscFem REQUIRED)
-find_package(SofaMiscCollision REQUIRED)
-find_package(SofaComponentAll REQUIRED)
-find_package(SceneCreator REQUIRED)
-find_package(Sofa.Testing REQUIRED)
+sofa_find_package(Sofa.Framework QUIET REQUIRED)
+sofa_find_package(Sofa.Component QUIET REQUIRED)
+sofa_find_package(SceneCreator QUIET REQUIRED)
+sofa_find_package(Sofa.Testing QUIET REQUIRED)
```
Comment (guparan) [inline applications/plugins/SofaTest/SofaTestConfig.cmake.in]:
```suggestion
find_package(Sofa.Framework QUIET REQUIRED)
find_package(Sofa.Component QUIET REQUIRED)
find_package(SceneCreator QUIET REQUIRED)
find_package(Sofa.Testing QUIET REQUIRED)
```

File: modules/CMakeLists.txt [hunk 4]
```diff
@@ -24,8 +24,8 @@ sofa_add_plugin(SofaExporter SofaExporter ON)
 
 sofa_add_plugin(SofaSparseSolver SofaSparseSolver ON)      #
 sofa_add_plugin(SofaPreconditioner SofaPreconditioner ON)  # depends on SofaSparseSolver
-sofa_add_plugin(SofaHaptics SofaHaptics ON)
-sofa_add_plugin(SofaValidation SofaValidation ON)
+#sofa_add_plugin(SofaHaptics SofaHaptics ON)
+#sofa_add_plugin(SofaValidation SofaValidation ON)
```
Comment (guparan) [inline modules/CMakeLists.txt]:
```suggestion
```

File: modules/SofaHaptics/SofaHapticsConfig.cmake.in [hunk 5]
```diff
@@ -3,7 +3,7 @@
 @PACKAGE_GUARD@
 @PACKAGE_INIT@
 
-find_package(SofaConstraint QUIET REQUIRED)
+find_package(Sofa.Component.Constraint.Lagrangian.Solver QUIET REQUIRED)
```
Comment (guparan) [inline modules/SofaHaptics/SofaHapticsConfig.cmake.in:7]:
```suggestion
find_package(Sofa.Simulation.Core QUIET REQUIRED)
find_package(Sofa.Component.Constraint.Lagrangian.Solver QUIET REQUIRED)
```

File: modules/SofaHaptics/SofaHaptics_test/scenes/ToolvsFloorCollision_test.scn [hunk 6]
```diff
@@ -1,13 +1,8 @@
 <?xml version="1.0" ?>
 <Node name="root" dt="0.05" showBoundingTree="0" gravity="0 -1 0">
-    <RequiredPlugin name="SofaOpenglVisual"/>
+    <RequiredPlugin name="Sofa.Component" />
```
Comment (guparan) [inline modules/SofaHaptics/SofaHaptics_test/scenes/ToolvsFloorCollision_test.scn:3]: I see you lazy boy :eyes:

**Review by guparan — CHANGES_REQUESTED** [review]
Here is a first screening. More may come.

File: modules/SofaHaptics/SofaHaptics_test/scenes/ToolvsFloorCollision_test.scn (hunk 6 above)
Comment (fredroy) [reply modules/SofaHaptics/SofaHaptics_test/scenes/ToolvsFloorCollision_test.scn:3]: 🫢 
it is a test so I dont really bother

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by alxbilger** [comment]
CI is happy: marked as ready

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
newly introduced tests not compatible with the compat, back to wip  😬

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Pull request merged**